app/controllers/shared/api.php

```diff
@@ -111,6 +109,18 @@
             throw new Exception(Exception::PROJECT_UNKNOWN);
         }
 
+        // Validate x-appwrite-timestamp header
+        $timestampHeader = $request->getHeader('x-appwrite-timestamp');
```

Comment:
We usually don't use `App::setResource` from inside a route / hook. We can set it on [init.php](https://github.com/appwrite/appwrite/blob/master/app/init.php) and still access the request header.

Reply:
@eldadfux are we able to do validation and throw an exception in `App::setResource()`?

Reply:
@eldadfux updated!